VTUL/dlp-access
# LIBTD-2338: Allow siteAdmin user to update filter and sort fields on …

…Browse Collections page

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2338) (:star:)

# What does this Pull Request do? (:star:)
This PR allows a siteAdmin user to update filter and sort configurations on the "Browse Collection" Page.

# What's the changes? (:star:)

* Added tests for add/update/delete filter field and add/delete sort fields
* Updated SortByDropdown component to allow for more sort fields
* Added BrowseCollectionsForm component to allow a siteAdmin user to update filter and sort fields as well as record the modification history
* Added a link to browseCollectionsForm on the siteAdmin page 

# How should this be tested?

A description of what steps someone could take to:
* Go to siteAdmin page then click on the "Filter and Sort Config for Browse Collections Page" link
* Try to add/update/delete filter values or add/delete sort fields, submit the form
* Go to "Browse Collections" page from Site main menu, check if the filter and sorting options are changes accordingly
* Check if the History table in DynamoDB records the modifications properly

# Interested parties
Tag (@yinlinchen) interested parties

(:star:) Required fields
